### PR TITLE
added tf state backend in azure

### PR DESCRIPTION
**Pull request by vcosqui**

### CODE REVIEW

File: azure/providers.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">=1.0"
+  backend "azurerm" {
+      resource_group_name  = "migrations"
+      storage_account_name = "migrationstfstateazure"
+      container_name       = "tfstate"
+      key                  = "terraform.tfstate"
```
Comment (ogomezso) [inline azure/providers.tf]: I would extract these setting to tf variables

**Review by ogomezso — CHANGES_REQUESTED** [review]

**Review by ogomezso — CHANGES_REQUESTED** [review]
I would extract the blob storage settings to tf variables

File: azure/providers.tf (hunk 1 above)
Comment (ogomezso) [reply azure/providers.tf]: And perhaps all the backend configuration to a backend.tf file so anyone can easily switch to local backend just avoiding this file

**Comment by vcosqui** [comment]
I am afraid it can't be done
apparently init phase does not have access to variables
https://stackoverflow.com/a/65010204

**Comment by ogomezso** [comment]
> I am afraid it can't be done apparently init phase does not have access to variables https://stackoverflow.com/a/65010204

So perhaps we can put it as plain env variables or even placeholders, I'm trying to avoid having references to specific resources on this repo.

Thanks!

**Review by ogomezso — APPROVED** [review]

**Review by ogomezso — APPROVED** [review]

**Pull request merged**